dali/base/dadiags.cpp

```diff
@@ -122,7 +123,7 @@ class CDaliDiagnosticsServer: public IDaliServer, public Thread
                     mb.append(getReceiveQueueDetails(buf).str());
                 }
                 else if (0 == stricmp(id, "locks")) {
-                    mb.append(querySDS().getLocks(buf).str());
+                    querySDS().getLocks(mb);
```

Comment:
Just FYi. In my code, the "locks" request from dalidiag DOES NOT go via getDaliDiagnosticValue(const char *name, StringBuffer &ret). It is from my getLockDataTreeIterator() which calls: getDaliDiagnosticValue(mb). dalidiag.cpp calls helper.formatLocks(itr, true, 0, ip, NULL, NULL, buf) for formatting.